graphql/schemabuilder/batch.go

```diff
@@ -312,30 +319,37 @@ func (funcCtx *batchFuncContext) prepareResolveArgs(sources []interface{}, args
 		in = append(in, reflect.ValueOf(selectionSet))
 	}
 
-	return in
+	return in, idxValues
 }
 
 // extractResultsAndErr converts the response from calling the function into
 // the expected type for the response object (as opposed to a reflect.Value).
 // It also handles reading whether the function ended with errors.
-func (funcCtx *batchFuncContext) extractResultsAndErr(numResps int, out []reflect.Value, retType graphql.Type) ([]interface{}, error) {
+func (funcCtx *batchFuncContext) extractResultsAndErr(out []reflect.Value, idxValues []reflect.Value, retType graphql.Type) ([]interface{}, error) {
```

Comment:
maybe i'm missing something, but the idxValues don't seem to be always reflect.ValueOf(0..n-1), so i think we don't have to do this list thing?

Reply:
Wanted to reduce the number of calls we made to `reflect.Value` as much as possible.  Since we already had to set the map keys in the provider function thought it would be more efficient to keep the idxValues around. 🤷‍♂ Seemed kinda weird to remake calls into something we just created a couple of lines before.

Reply:
Also didn't like passing in the length attribute, it felt odd.